tests/integration_cases/test_membership_changes.rs

```diff
@@ -76,16 +78,16 @@ mod api {
     #[test]
     fn checks_for_voter_demotion() -> Result<()> {
         setup_for_test();
-        let mut raft = Raft::new(
-            &Config {
-                id: 1,
-                tag: "1".into(),
-                peers: vec![1, 2, 3],
-                learners: vec![4],
-                ..Default::default()
-            },
-            MemStorage::new(),
-        )?;
+        let config = Config {
+            id: 1,
+            tag: "1".into(),
+            peers: vec![1, 2, 3],
+            learners: vec![4],
+            ..Default::default()
+        };
+        let store = MemStorage::new();
+        store.initialize_with_config(&config);
+        let mut raft = RawNode::new(&config, store)?.raft;
```

Comment:
This seems like a pretty complicated way to get a `raft` object. Does this only affect tests or should we add some convenience API?